src/Spectre/Controllers/PreparationsController.cs

```diff
@@ -74,5 +77,43 @@ public Spectrum Get(int id, int spectrumId)
 
             return new Spectrum() { Id = spectrumId, Intensities = intensities, Mz = mz, X = coordinates.X, Y = coordinates.Y };
         }
+
+        /// <summary>
+        /// Gets single heatmap of a specified preparation based on provided mz.
+        /// </summary>
+        /// <param name="id">Preparation identifier.</param>
+        /// <param name="mz">Mz value serving as heatmap basis.</param>
+        /// <returns>Heatmap</returns>
+        /// <exception cref="ArgumentException">Thrown when provided mz is lower 
+        /// than zero, or is invalid for a given dataset</exception>
+        public Heatmap Get(int id, double mz)
+        {
+            if (id != 1)
+                return null;
+
+            if (mz < 0.0)
+                throw new ArgumentException("Mz lower than zero provided");
+
+            IDataset dataset = new BasicTextDataset("C:\\spectre_data\\hnc1_tumor.txt");
+
+            int indexOfMz = dataset.GetRawMzArray().ToList().IndexOf(mz);
+
+            if(indexOfMz == -1)
+                throw new ArgumentException("Provided mz value wasn't found in the dataset");
+
+            var intensities = dataset.GetRawIntensityRow(indexOfMz);
+            var coordinates = dataset.GetRawSpacialCoordinates(true);
+
+            int[] xCoordinates = new int[intensities.Length];
+            int[] yCoordinates = new int[intensities.Length];
+
+            for (int i = 0; i < intensities.Length; i++)
+            {
+                xCoordinates[i] = coordinates[i, 0];
+                yCoordinates[i] = coordinates[i, 1];
+            }
```

Comment:
This will have to be extracted to a service at some point, but for now it's ok (as for mock).